### PR TITLE
Move repo modification jobs to GitHub actions

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
This moves the repository modification jobs for the release-notes,
dependencies as well as the branch fast-forward to GitHub actions.

#### Which issue(s) this PR fixes:

Refers to #4489 
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: scripts/dependencies/dependencies.go [hunk 1]
```diff
@@ -115,8 +116,14 @@ _Generated on %s for commit [%s][0]._
 		return errors.Wrap(err, "writing report")
 	}
 
-	// Update gh-pages branch if not a pull request and running in CircleCI
-	if util.IsEnvSet("CIRCLECI") && !util.IsEnvSet("CIRCLE_PULL_REQUEST") {
+	// Update gh-pages branch if not a pull request and running in CI
+	token, tokenSet := os.LookupEnv(tokenKey)
+	if !tokenSet || token == "" {
+		logrus.Infof("%s environemt variable is not set", tokenKey)
```
Comment (haircommander) [inline scripts/dependencies/dependencies.go:122]: nit: environment*

**Comment by haircommander** [comment]
any reason we did the krel version bump here too?

File: scripts/dependencies/dependencies.go (hunk 1 above)
Comment (saschagrunert) [reply scripts/dependencies/dependencies.go:122]: Fixed :+1:

**Comment by saschagrunert** [comment]
> any reason we did the krel version bump here too?

We improved the release notes fetching logic (there was this issue that commands get stuck), so I thought this is a good point where we should do the update. Should I revert that change?

**Comment by haircommander** [comment]
no that's fine :) 

/test integration_fedora
/lgtm

**Pull request merged**